ragatouille/data/training_data_processor.py

```diff
@@ -137,8 +137,8 @@ def _process_raw_pairs(self, raw_data, mine_hard_negatives, n_new_negatives):
 
         for query, positive in raw_data:
             if isinstance(positive, str):
-                positive = [positive]
-            raw_grouped_triplets[query]["positives"] += positive
+                raise ValueError
+            raw_grouped_triplets[query]["positives"].append(positive['content'])
```

Comment:
I think this should keep the current logic, but the current pre-processing is a problem.

Ideally, we'd want to either:
- Modify the corpus splitter so that it never returns dictionaries but rather strings (but in this case we'd need to make changes in other places where the `"content"` key is accommodated for, so probably something for later)
- Make it so an earlier stage in the data processing should check if anything is a dict and extract the `content` key to a string beforehand...

This is because ideally this step should accept either strings or list of strings (`(query: str, positives: list[str])`) is a viable input), so erroring out and assuming that people have used our (flawed) data processor isn't ideal behaviour.